src/main/java/Parser.java

```diff
@@ -0,0 +1,186 @@
+
+        try {
+            if (fullCommand.equals("")) {
+                throw new NullPointerException("");
+            }
+
+            switch (splitBySpace[0]) {
+            case "list":
+                return Command.LIST;
+            case "done":
+                return Command.DONE;
+            case "delete":
+                return Command.DELETE;
+            case "todo":
+                return Command.TODO;
+            case "event":
+                return Command.EVENT;
+            case "deadline":
+                return Command.DEADLINE;
+            case "find":
+                return Command.FIND;
+            default:
+                throw new DukeException("Don't understand");
+            }
+        } catch (DukeException e) {
+            System.out.println(e);
+        } catch (NullPointerException e) {
+            System.out.println("Please enter a command.");
+        }
+
+        return null;
+    }
+
+    /**
+     * This function will parse the description of todo command.
+     * @param fullCommand Command that the user typed.
+     * @return The description of todo command.
+     */
+    public String parseDescription(String fullCommand) {
+        String[] splitBySpace;
+        splitBySpace = fullCommand.split(" ");
+        String newString = "";
+
+        try {
+            if (splitBySpace.length == 1) {
+                throw new DukeException(splitBySpace[0]);
+            }
+
+            for (int i = 1; i < splitBySpace.length - 1; i++) {
+                newString = newString + splitBySpace[i] + " ";
+            }
+            newString = newString + splitBySpace[splitBySpace.length - 1];
+
+            return newString;
+        } catch (DukeException e) {
+            System.out.println(e);
+        }
+
+        return null;
+    }
+
+    /**
+     * This function will parse the description of event and deadline command.
+     * @param fullCommand Command that the user typed.
+     * @return Description of event and deadline command.
+     */
+    public String parseDescOfEventDeadline(String fullCommand) {
+        String[] splitBySpace;
+        String[] splitBySlash;
+        splitBySpace = fullCommand.split(" ");
+        String newString = "";
+
+        try {
+            if (splitBySpace.length == 1) {
+                throw new DukeException(splitBySpace[0]);
+            }
+
+            splitBySlash = fullCommand.split("/");
+            String[] splitBySpace2 = splitBySlash[0].split(" ");
+
+            for (int i = 1; i < splitBySpace2.length - 1; i++) {
+                newString = newString + splitBySpace2[i] + " ";
+            }
+            newString = newString + splitBySpace2[splitBySpace2.length - 1];
+
+            return newString;
+
+        } catch (DukeException e) {
+            System.out.println(e);
+        }
+
+        return null;
+    }
+
+    /**
+     * This function will parse the "deadline" typed by the user. Function will only be called if the command is
```

Comment:
Can just write "Parses the..." rather than "This function..." for your javadocs.